content/hardware/06.nicla/boards/nicla-voice/tutorials/ei-intruder-detector/content.md

```diff
@@ -0,0 +1,474 @@
+  } else if (!batteryLevelChar.subscribe() || !alertLevel.subscribe()) {
+    Serial.println("subscription failed!");
+    peripheral.disconnect();
+    return;
+  } else {
+    Serial.println("Subscribed to Battery Level and Alert Characteristic");
+  }
+```
+Finally, the Portenta verifies continuously if a characteristic is updated to upload it to the Cloud.
+
+```arduino
+// while the peripheral is connected
+  while (peripheral.connected()) {
+    
+    // check if the value of the characteristic has been updated
+    if (batteryLevelChar.valueUpdated()) {
+
+      batteryLevelChar.readValue(BatteryValue);
+
+      Serial.print("Battery Level: ");
+      Serial.print(BatteryValue);
+      Serial.println(" %");
+
+      battery = BatteryValue;
+    }
+
+    // check if the value of the characteristic has been updated
+    if (alertLevel.valueUpdated()) {
+
+      alertLevel.readValue(AlertValue);
+
+      Serial.print("Alert: ");
+      Serial.println(AlertValue);
+
+      
+      if (AlertValue == 1) {    // if the Alert = 1 means the door was opened
+        control = !control;     // as variables are updated if they change, add a simple dot "." to be able to update it if happens twice consecutive.
+        if (control) {
+          DoorEvent = "Door opened";
+        } else {
+          DoorEvent = "Door opened.";
+        }
+
+        alertStatus = false;
+
+      } else if (AlertValue == 2) {   // if the Alert = 2 means the door was forced
+        control = !control;           // as variables are updated if they change, add a simple dot "." to be able to update it if happens twice consecutive.
+        if (control) {
+          DoorEvent = "Intruder detected!";
+        } else {
+          DoorEvent = "Intruder detected!.";
+        }
+
+        alertStatus = true;
+      }
+    }
+
+    // turn off green LED if disconnected from Cloud.
+    if (ArduinoCloud.connected()) {
+      digitalWrite(LEDG, LOW);
+    } else {
+      digitalWrite(LEDG, HIGH);
+    }
+
+    ArduinoCloud.update();
+  }
+```
+### The Cloud Dashboard
+
+Taking advantage of the Arduino Cloud, we can seamlessly integrate a simple but powerful dashboard to monitor and visualize the status of the system in real-time as you can see below:
+
+![Arduino Cloud project dashboard ](assets/dashboard.png)
+
+Within the Arduino Cloud's dashboard, the system variables can be monitored, we have a battery level indicator widget accompanied by a time series graph, a chat-looking widget to store the events historically, a red LED to shine when an intruder is detected by the system, and a green LED to show the BLE connection status between the Nicla Voice and the Portenta H7 host. We can easily access this dashboard from a PC, mobile phone or tablet from anywhere, receiving an instantaneous update wherever we are. In addition, we can set different integrations to complement our project, for example, setting up an IFTTT automation to receive an email whenever an alert is fired.
```

Comment:
```suggestion
Within the Arduino Cloud's dashboard, the system variables can be monitored. We have a battery level indicator widget accompanied by a time series graph, a chat-looking widget to store the events historically, a red LED to shine when an intruder is detected by the system, and a green LED to show the BLE connection status between the Nicla Voice and the Portenta H7 host. We can easily access this dashboard from a PC, mobile phone or tablet from anywhere, receiving an instantaneous update wherever we are. In addition, we can set different integrations to complement our project, for example, setting up an IFTTT automation to receive an email, an cellphone notification, or even triggering some automation like turning on all the lights at home whenever an alert is fired.
```

Your first sentences are always great and explain the main idea in just a few words. They are full meaning so we can put an stop there to avoid longer phrases.

Some suggestions, the good thing about an IFTTT service is that you can connect things to things in a very easy way, we can even start to play some music on smart speakers to give the thief the impression that someone is at home.